gated-infrastructure-deploy/README.md

```diff
@@ -43,26 +55,88 @@ This template makes use of `terraform plan` and `terraform apply`.
+    | parameter | Required | Type | Description | 
+    | - | - | - | - | 
+    | AzDOEnvironmentName | Yes | string | Environment name as defined in Azure DevOps |
+    | TFStateResourceGroupName | Yes | string | Terraform state resource group |
+    | TFStateStorageAccountName | Yes | string | Terraform state storage account |
+    | TFStateContainerName | Yes | string | Terraform state container |
+    | TFStateBlobName | Yes| string | Terraform state blob |
+    | TerraformWorkspace | Yes | string | Terraform workspace |
+    | TerraformArtifact| Yes | string | Artifact containing your .tf files and any other supporting files for your deployment |
+    | TerraformArtifactConfigRelativePath | Yes | string | Relative path to the .tf files inside your artifact |
+    | VariablesTemplateRelativePath | Yes | string | Relative path to a YAML template containing a variables expression |
+    | TerraformVariableMappings | Yes | object | A key/value map of Terraform variables to be injected into the PowerShell runtime environment for Terraform to use |
+    | TerraformOutputVariables| No | object | An array of Terraform output variables to be retrieved after the Terraform apply has completed |
 
-## How does it work?
+    **Notes**:
 
-The template is broken into three jobs:
+    - *TerraformWorkspace*: This value will be appended onto the blob name in the form `[TFStateBlobName]:[TerraformWorkspace]`, e.g. `terraform.deployment.tfplan:dev`   
 
-- the plan job
-- the manual verification
-- the apply job
+    - *TerraformArtifact*: The files in the artifact will be used without modifying their contents. Your pipeline will need to include jobs to make any required modfications to your supporting files e.g. inserting configuration values, before publishing the artifact.
+
+    - *TerraformArtifactConfigRelativePath*: Inside the template, the full path to the configuration will be `$(Pipeline.Workspace)/[TerraformArtifact][TerraformArtifactConfigRelativePath]`.
+      
+      Example 1: The artifact is named 'tfartifact' and the .tf files are in the root. `TerraformArtifactConfigRelativePath` would be `/` and the concatenated path would be `$(Pipeline.Workspace)/terraformartifact/`.
+
+      ```
+      +-- terraform
+      |   +-- azure.tf
+      |   +-- main.tf
+      |   +-- variables.tf
+      |   +-- output.tf
+      ```
```

Comment:
```suggestion
      ```
      +-- tfartifact
      |   +-- azure.tf
      |   +-- main.tf
      |   +-- variables.tf
      |   +-- output.tf
      ```
```

Aligning with the example.